nestkernel/simulation_manager.h

```diff
@@ -146,6 +146,24 @@ class SimulationManager : public ManagerInterface
   // TODO: Precisely how defined? Rename!
   Time const& get_clock() const;
 
+  /**
+   * Get the number of total cycles in the current run.
+   * Can be used to calculate the simulation time of the current run.
+   */
+  delay get_number_of_steps_in_run() const;
+
+  /**
+   * Get the start cycle of the current run.
+   * Can be used to calculate the start time of the current run.
+   */
+  delay get_start_step_in_run() const;
+
+  /**
+   * Get the number of cycles that the current simulation will have been simulated for.
+   * Can be used to calculate the end time of the simulation (only considering the runs that have be called so far).
+   */
+  delay get_end_step_in_simulation() const;
```

Comment:
Why is this method called `..._simulation()` while the other two are called `..._run()`? Shouldn't they be consistent?

Reply:
Compared to the other two functions, which will return the start/end time in their respective run, this function returns the end time over multiple runs that were started in the same RunManager (e.g. when calling Run(10) three times then the _simulation function would return 10/20/30 when being called during the respective runs). I thought of the different runs which are in the same RunManager as "the simulation" and thus named the function ..._simulation(). The other two return information that are local to their respective run.

I have to admit that I wasn't 100% sure what names would align the best with the rest of the NEST codebase to model this distinction between a run and what I assumed to be "the overall simulation".

Reply:
The `get_start_...()` function also returns the "full" time at the beginning of each `run()`,  call, in your example above 0, 100 and 300, i.e., it is always the sum of the total number of steps completed in the past. In the same way, `get_end_...()` returns the corresponding end times, 100, 300 and 600. So the `get_start_...()` and `get_end_...()` methods are symmetrical and should have the corresponding names; I also think the correct preposition here is "of".

I will add a suggestion for changes in the comment and code below. Corresponding name changes are required in the definition of the methods.